crates/lib/src/action/workingdata.rs

```diff
@@ -116,31 +120,98 @@ impl Default for WorkingData {
 ///
 /// The [`Action::outcome()`] method is the only way to set the outcome of the action, and it _must_
 /// be called before the handler returns.
-#[derive(Debug)]
 pub struct Action {
 	/// The collected events which triggered the action.
 	pub events: Arc<[Event]>,
-	pub(super) outcome: Arc<OnceCell<Outcome>>,
+	processes: Arc<[SupervisorId]>,
+	pub(crate) outcomes: Arc<Mutex<HashMap<SupervisorId, (Resolution, EventSet)>>>,
+}
+
+impl std::fmt::Debug for Action {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		f.debug_struct("Action")
+			.field("events", &self.events)
+			.field("processes", &self.processes)
+			.finish()
+	}
 }
 
 impl Action {
-	pub(super) fn new(events: Arc<[Event]>) -> Self {
+	pub(crate) fn new(events: Arc<[Event]>, processes: Arc<[SupervisorId]>) -> Self {
 		Self {
 			events,
-			outcome: Default::default(),
+			processes,
+			outcomes: Default::default(),
 		}
 	}
 
-	/// Set the action's outcome.
+	/// Set the action's outcome for all [`Process`]es.
 	///
 	/// This takes `self` and `Action` is not `Clone`, so it's only possible to call it once.
 	/// Regardless, if you _do_ manage to call it twice, it will do nothing beyond the first call.
```

Comment:
oh, you are right! 

Would something like:
```rust
/// Set the action's outcome for all [`Process`]es that were active when the action was created.
```
be more accurate?

Reply:
Oh, I didn't mean that, I meant the `if you _do_ manage to call it twice, it will do nothing beyond the first call.` part.

Reply:
hmm.. yess, right! because it does change it to the second one now, good point! should this behavior be kept and the comment adjusted to fit, or the other way around, in your opinion?

Reply:
Ah, just remove that line then. It shouldn't be callable twice anyway, and if a user hacks around to do that on purpose, it will be their fault if they get surprised.

Reply:
haha, fair! if you do crazy things, expect crazy results! :D

 how do you reckon one could hack around it, if you think it is possible at all?

Reply:
transmutes, probably